src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,116 @@
+import os
+import tempfile
+import unittest
+import unittest.mock
+
+import pytest
+
+import insights_client
+
+
+class MockFS:
+    """Mock filesystem.
+
+    Ensures that the files required for the MOTD manipulation are created
+    in a temporary directory instead of working with actual system files.
+    """
+
+    def __init__(self):
+        self.dir = tempfile.TemporaryDirectory()
+
+        # /etc/insights-client/insights-client.motd
+        self.dir_src = unittest.mock.patch(
+            "insights_client.MOTD_SRC", f"{self.dir.name}/src"
```

Comment:
For safe path joining, use os.path.join instead of string interpolation.

```suggestion
            "insights_client.MOTD_SRC", os.path.join(f"{self.dir.name}/src"
```